### DISCUSSION TITLE
Unable to build package in a local environment

### DISCUSSION THREAD

**Discussion by aegroto** [stackernews/stacker.news#1675] [local dev help]
It would be beneficial to be able to build the package in a local environment. For instance, PRs such as #1606 require to verify that the bundle size is not affected too much. However, running the start ```npm run build``` command fails in the current project structure. I am currently using this Dockerfile:

```Dockerfile
# syntax=docker/dockerfile:1

FROM node:18.20.4-bullseye

ENV NODE_ENV=production

WORKDIR /app

EXPOSE 3000

COPY . .
RUN npm ci --legacy-peer-deps --loglevel verbose
RUN npm run build --verbose
```

The command ```docker build -t sn-test --no-cache .``` fails with a bunch of the following errors:

```
unhandledRejection TypeError [ERR_INVALID_URL]: Invalid URL
    at new NodeError (node:internal/errors:405:5)
    at new URL (node:internal/url:676:13)
    at /app/.next/server/chunks/4926.js:1:43211
    at 69496 (/app/.next/server/chunks/4926.js:1:43728)
    at t (/app/.next/server/webpack-runtime.js:1:127)
    at /app/.next/server/chunks/4926.js:1015:5058
    at t.a (/app/.next/server/webpack-runtime.js:1:866)
    at 95639 (/app/.next/server/chunks/4926.js:1015:4951)
    at t (/app/.next/server/webpack-runtime.js:1:127)
    at /app/.next/server/chunks/4926.js:3986:461 {
  type: 'TypeError',
  input: 'undefined',
  code: 'ERR_INVALID_URL'
}
```

I have uploaded the full logs and the update Dockerfiles in https://github.com/aegroto/stacker.news/tree/local-bundle-build.

I have been looking for a while on the web and it looks like these errors may be due to missing or wrongly formatted environment variables pointing to URLs. However I have manually checked and the .env files are copied in the resulting image, therefore the environment should be set up correctly. Is there any other step missing? Thanks in advance.

**Comment by huumn** [comment]
Did you copy `.env.development` -> `.env.production`? When `NODE_ENV = 'production'`, it only uses `.env.production`.

**Reply by aegroto** [reply]
Thanks for your quick feedback. Do you mean that you actually need the variabiles in both `.env.development` and `.env.production` to be set? I thought only of the envs is supposed to be activated.

**Reply by huumn** [reply]
`.env.production` does not have all the production env variables in it because many of them are secrets

**Comment by aegroto** [comment]
I have been able to build by creating a `.env` file with variables from both `development` and `production`, adding VAPID_ variables as indicated [here](https://github.com/stackernews/stacker.news?tab=readme-ov-file#enabling-web-push-notifications) with a subject such as `mailto:aegroto@stacker.news`, then using this Dockerfile:

```
# syntax=docker/dockerfile:1

FROM node:18.20.4-bullseye

WORKDIR /app

EXPOSE 3000

COPY . .
RUN npm ci --legacy-peer-deps --loglevel verbose
RUN npm run build --verbose
CMD ["/bin/bash"]
```

There is no need to set `NODE_ENV`. the `.env` file is loaded automatically.

I will work on converting this to work in a GitHub action such that we can perform checks on the bundle directly during the CI pipeline.

**Reply by huumn** [reply]
Awesome!
